### ISSUE TITLE
docs: add training section

### ISSUE THREAD

**Issue by dayanchm** [dayanchm/Orvilo#3]
Create and expand the `training/` section.

Topics may include:
- Pre-training
- Fine-tuning
- Instruction tuning
- Loss functions
- Backpropagation
- RLHF
- Reinforcement learning
- Distillation
- Synthetic data
